File: 2515.ShortestDistancetoTargetStringinaCircularArray.cpp
//submission one
class Solution {
public:
    int closestTarget(vector<string>& words, string target, int startIndex) {
        int n = words.size();
        int ans = INT_MAX;
        
        for(int i = 0; i<n; i++){
            if(words[i] == target){
                
                int front = (i - startIndex + n)%n;
                int back = (startIndex - i + n)%n;
            
                ans = min(ans, min(front, back));
                
            }
            
        }
        
        return ans == INT_MAX ? -1:ans;
    }
};

File: 2515.ShortestDistancetoTargetStringinaCircularArray.py
//submission two

class Solution:
    def closestTarget(self, words: List[str], target: str, startIndex: int) -> int:
        n = len(words)
        ans = math.inf
        
        for i in range(n):
            if(words[i]==target):
                forward = (i-startIndex+n)%n
                backward= (startIndex-i+n)%n
                
                ans = min(ans, min(forward, backward))
                
        
        if(ans == math.inf):
            return -1
        else:
            return ans